paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/KYCRefreshScreen.kt

```diff
@@ -0,0 +1,228 @@
+                        ),
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .height(LinkTheme.shapes.primaryButtonHeight)
+                    ) {
+                        Text(
+                            "Confirm",
+                            style = LinkTheme.typography.body.copy(fontWeight = FontWeight.SemiBold)
+                        )
+                    }
+                }
+            }
+        }
+    }
+}
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+interface VerifyKYCInfo {
+    val firstName: String
+    val lastName: String
+    val dateOfBirth: DateOfBirth
+    val idNumberLastFour: String?
+    val address: PaymentSheet.Address
+}
+
+@Composable
+private fun TopNavigationBar(
+    onClose: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Row(
+        modifier = modifier
+            .fillMaxWidth()
+            .padding(vertical = 8.dp),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Icon(
+            painter = painterResource(id = getLinkIcon()),
+            contentDescription = "Link",
+            tint = Color.Unspecified,
+            modifier = Modifier
+                .height(72.dp)
+                .width(88.dp)
+        )
+
+        Spacer(modifier = Modifier.weight(1f))
+
+        IconButton(
+            onClick = onClose,
+        ) {
+            Box(
+                modifier = Modifier
+                    .size(36.dp)
+                    .clip(CircleShape)
+                    .background(LinkTheme.colors.textPrimary.copy(alpha = 0.12f))
+            ) {
+                Icon(
+                    painter = painterResource(R.drawable.stripe_ic_paymentsheet_close),
+                    contentDescription = "Close",
+                    tint = LinkTheme.colors.textPrimary,
+                    modifier = Modifier.align(Alignment.Center)
+                        .size(12.dp)
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun InfoRow(
+    title: String,
+    value: String,
+    icon: (@Composable () -> Unit)? = null,
+    onIconTap: (() -> Unit)? = null
+) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(16.dp),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Column(modifier = Modifier.weight(1f)) {
+            Text(
+                text = title,
+                style = LinkTheme.typography.caption,
+                color = LinkTheme.colors.textSecondary
+            )
+            Text(
+                text = value,
+                style = LinkTheme.typography.body,
+                color = LinkTheme.colors.textPrimary
+            )
+        }
+        icon?.let {
+            IconButton(onClick = onIconTap ?: {}) {
+                it()
+            }
+        }
+    }
+}
+
+private fun PaymentSheet.Address.formattedAddress(): String {
+    return listOf(
```

Comment:
How do we format addresses elsewhere? I feel like this should already exist.

Reply:
I thought so as well, but the closest I could find was in `AddressElementExampleActivity`, and it's not quite the same. Happy to be proven wrong though.